src/NumberGuessingGame.js

```diff
@@ -1,77 +1,79 @@
-import React, { Component } from "react";
+import React, { useState } from "react";
 import GuessControl from "./GuessControl";
 import GuessMessage from "./GuessMessage";
 import GameOver from "./GameOver";
 
-/**
- *
- * Returns a random integer number from 1-100 inclusive
- */
 function getRandomNumber() {
   return Math.floor(Math.random() * 100) + 1;
 }
 
-const MAX_ATTEMPTS = 5;
+const MAX_ATTEMPTS = 4;
 
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
 
-    this.state = {
-      numberToGuess: getRandomNumber(),
-      numberOfGuesses: 0,
-      latestGuess: null,
-    };
+export default function NumberGuessingGame  () {
+  const [currentGuess, setCurrentGuess] = useState("");
+  const [numberOfGuesses, setNumberOfGuesses] = useState(0);
+  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber());
```

Comment:
You have a bug here --- I actually stumbled upon the same bug 😆 

`console.log(numberToGuess)` and you'll see the initialize `numberToGuess` is different than the number you guess after your first try 😮 . React is weird. But there's a way to get around that and assign `getRandomNumber` to `numberToGuess`

Reply:
💯 on fixing the bug